js/src/util/args.ts

```diff
@@ -21,11 +21,26 @@ import { Column } from '../column';
 import { Vector } from '../vector';
 import { DataType } from '../type';
 import { Chunked } from '../vector/chunked';
+import { BigIntArray, TypedArray } from '../interfaces';
+import { FloatVector, IntVector } from '../vector/index';
 
 type RecordBatchCtor = typeof import('../recordbatch').RecordBatch;
 
 const isArray = Array.isArray;
 
+/** @ignore */
+export function isTypedArray(arr: any): arr is TypedArray | BigIntArray {
+    return ArrayBuffer.isView(arr) && 'BYTES_PER_ELEMENT' in arr;
+}
+
+/** @ignore */
+function vectorFromTypedArray(array: TypedArray): Vector {
+    if (array instanceof Float32Array || array instanceof Float64Array) {
+        return FloatVector.from(array);
+    }
+    return IntVector.from(array);
```

Comment:
Use Vector.new with Data.new here.

Reply:
@trxcllnt I took your feedback but this feels overly verbose and also now I expose a previously internal function. 

<img width="855" alt="Screen Shot 2021-04-26 at 17 33 23" src="https://user-images.githubusercontent.com/589034/116167669-86746b00-a6b5-11eb-812f-103f0daaf67f.png">

Reply:
Thanks for the help on Slack. I updated the pull request.